src/Collection/Iterator/MapIterator.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Munus\Collection\Iterator;
+
+use Munus\Collection\Iterator;
+use Munus\Exception\NoSuchElementException;
+use Munus\Tuple;
+
+final class MapIterator extends Iterator
```

Comment:
Maybe `Iterator` should be an interface. Best option for this is interface with `default` implementation, but this is not Java :disappointed: